lib/utilities/schedule.cpp

```diff
@@ -0,0 +1,50 @@
+#include "schedule.h"
+
+int32_t Schedule::index = 0;
+
+Schedule::Schedule(time_t start, float rate, int32_t minutes)
+:rate(rate),minutes(minutes),start(start)
+{
+  index++;
```

Comment:
This is pretty nasty, is there really no other way to achieve this?
Instead of having this in the class you might want to inject this within the constructor.